### DISCUSSION TITLE
Filter on entries containing a Unicode white space

### DISCUSSION THREAD

**Discussion by clrudolphi** [FreshRSS/FreshRSS#6051] [Q&A]
I'm self hosting. Everything works just fine but I am trying to fine-tune my experience. I read on mobile thru FeedMe.

I have a feed from a news organization. I'd like to see only those articles within that feed that cover a small set of topics. There is no way to filter via the feed directly (such as with query parameters on the feed url).

I would like to have FreshRss automatically mark as read articles within that feed which do not contain any of my multiple keywords. I'm using the subscription management page for the feed and enter keyword filters into the "Mark as Read..." section.

I've tried them all on one line such as:
!Topic1 !Topic2 !"multi word topic3"

I've also tried putting each topic on separate lines.

Is what I want possible?
Am I in the right spot for configuring this?
Which keyword search/filter operators should I be using? (Is negation the correct approach)
Should these be on a single line or multiple?

My experience so far is that all topics (negated) filters nothing (everything continues to show up) while each on separate lines marks everything as read.

If anyone has a tutorial already available please point me there.
TIA

**Comment by Alkarex** [comment]
Try on one line something like:
```
!(want OR also-want OR "also want")
```

**Reply by clrudolphi** [reply]
Thanks for the suggestion.

When I put each negated term on a separate line then all articles in the feed are marked as read.
When I put all keywords on a single line, each individually negated and separated by a space then nothing gets marked as read.
When using your suggestion of a set of terms OR'd together and then negated, I get the same behavior as in the first attempt (everything is marked as read, even articles that include one or more keywords).

Are there other options to try? 
Are there logs that we could examine?
Should I be trying simple test cases first or with other feeds?

Thanks

**Reply by Alkarex** [reply]
Here is an example:
* Feed URL: https://github.com/FreshRSS/FreshRSS/releases.atom
* Mark an article as read: `!(Hungarian OR Dutch OR Czech)`
* Result:
![image](https://github.com/FreshRSS/FreshRSS/assets/1008324/568f8bf4-a899-4a62-90b0-7596c5bd8be2)

When using multiple lines, if any line matches, then the article is marked as read. It is equivalent to combining everything on the same line with an `OR`.

See also the documentation https://freshrss.github.io/FreshRSS/en/users/10_filter.html#with-the-search-field

You are welcome to show the feed and filter you are trying.

**Reply by clrudolphi** [reply]
This did not work when I put the query expression directly into the feed's "Mark an article as read..." field 

It did work when I created a user query using the same expression and then referenced that query in the "Mark an article as read..." field (using search:myFilterquery) 

One would think the two approaches should be equivalent?

It's working now for me. Thank you for your help.

**Reply by Alkarex** [reply]
Yes, the two approaches are equivalent

**Reply by Alkarex** [reply]
For more debugging, it would be good to see the queries.
In particular, do you have any special character?

**Reply by clrudolphi** [reply]
The feed URL is https://www.shawlocal.com/arcio/rss/

This is a feed from a publisher of small-town newspapers in northern Ilinois, USA. My intent is to keep only those articles that mention specific towns.
My saved query is: !(Dixon OR LaSalle OR Peru OR Lee OR Utica OR Starved OR Barrington OR Inverness OR Palatine)

This query, when referenced as a saved search query, works at filtering to articles about those towns. The same query when present in the Mark As Read box doesn't work.

**Reply by Alwaysin** [reply]
> This query, when referenced as a saved search query, works at filtering to articles about those towns. The same query when present in the Mark As Read box doesn't work.

@Alkarex may this be a bug? I think I'm hitting something similar, eg. on `https://www.lesnumeriques.com/rss.xml`, I had inputed in `Mark an article as read…` : `intitle:"reconditionnées ?"` and I still see the article published on 31 March 2024 at 18:00 `Guide d&#039;achat : Quelles sont les meilleures enceintes portables reconditionnées ?` (maybe there is another problem with the `&#039;` but it is not the subject here), it hasn't been marked as read, although when I search with the exact same query, it finds the article. I thought about maybe the ? character, because I have a lot of other intitle: that are correctly working.

**Reply by Alkarex** [reply]
@Alwaysin At a quick glance, in `Quelles sont les meilleures enceintes portables reconditionnées ?`, the space before the question mark seems to be a *Non-breaking space* (`C2A0` in UTF-8). Depending on your database (which one are you using?), this might be equal to a normal space (I quickly tested with SQLite where they are not equal, and with PostgreSQL where they seem to be equal).
In FreshRSS, the global search is using the SQL engine, while the auto mark as read is using PHP, and there might be subtle differences in *collation*.
Try with https://example.net/i/?search=intitle%3A%22reconditionn%C3%A9es%C2%A0%3F%22

**Reply by Alwaysin** [reply]
There doesn't seem to be a difference in global search between ```search=intitle%3A"reconditionnées%C2%A0%3F"``` and ```search=intitle%3A"reconditionnées+%3F"``` for MariaDB 10.11 with utf8mb4_general_ci at database level, utf8mb4_unicode_ci at table level. I didn't find how to check collation in PHP by itself. How do I input the search with the non-breaking space in the auto mark as read? Directly with the encoding?

**Reply by Alkarex** [reply]
> How do I input the search with the non-breaking space in the auto mark as read?

I believe copy pasting the Unicode character should work, if the Web browser does not mess it up. To be tested...

**Reply by Frenzie** [reply]
Normally that works, yes. There's also the Windows Character Map (since at least 3.11!), KCharSelect etc. where you can search for "no-break space" and copy it.
